### ISSUE TITLE
cgo build fail

### ISSUE THREAD

**Issue by mintzhao** [bazelbuild/rules_go#514]
### verison
bazel 0.5.1 
rule-go 0.4.4

```
ERROR: /home/travis/.cache/bazel/_bazel_travis/ef1dc36cffa27d24eeff815ad32c70b0/external/com_github_mattn_go_sqlite3/BUILD:5:1: in _cgo_filter_srcs rule @com_github_mattn_go_sqlite3//:cgo_default_library_filter_cgo_srcs: 
Traceback (most recent call last):
	File "/home/travis/.cache/bazel/_bazel_travis/ef1dc36cffa27d24eeff815ad32c70b0/external/com_github_mattn_go_sqlite3/BUILD", line 5
		_cgo_filter_srcs(name = 'cgo_default_library_filter_cgo_srcs')
	File "/home/travis/.cache/bazel/_bazel_travis/ef1dc36cffa27d24eeff815ad32c70b0/external/io_bazel_rules_go/go/def.bzl", line 850, in _cgo_filter_srcs_impl
		_emit_generate_params_action(cmds, ctx, script_name)
	File "/home/travis/.cache/bazel/_bazel_travis/ef1dc36cffa27d24eeff815ad32c70b0/external/io_bazel_rules_go/go/def.bzl", line 177, in _emit_generate_params_action
		ctx.new_file(ctx.configuration.bin_dir, fn)
Output artifact '/cgo_default_library_filter_cgo_srcs.CGoFilterSrcs.params' not under package directory 'external/com_github_mattn_go_sqlite3' for target '@com_github_mattn_go_sqlite3//:cgo_default_library_filter_cgo_srcs'.
```

**Comment by mintzhao** [comment]
```
INFO: Found 6 targets...
ERROR: /Users/mintzhao/Documents/gopath/src/github.com/mintzhao/yoyo/vendor/golang.org/x/sys/unix/BUILD.bazel:3:1: error executing shell command: 'bazel-out/local-fastbuild/bin/vendor/golang.org/x/sys/unix/bazel-out/local-fastbuild/genfiles/vendor/golang.org/x/sys/unix/cgo_default_library.cgo.dir.CGoCodeGenFile.params' failed: bash failed: error executing command 
  (exec env - \
    CGO_LDFLAGS='' \
    GOARCH=amd64 \
    GOOS=darwin \
  /bin/bash -c bazel-out/local-fastbuild/bin/vendor/golang.org/x/sys/unix/bazel-out/local-fastbuild/genfiles/vendor/golang.org/x/sys/unix/cgo_default_library.cgo.dir.CGoCodeGenFile.params)

Use --sandbox_debug to see verbose messages from the sandbox.
usage: cgo -- [compiler options] file.go ...
  -debug-define
        print relevant #defines
  -debug-gcc
        print gcc invocations
  -dynimport string
        if non-empty, print dynamic import data for that file
  -dynlinker
        record dynamic linker information in -dynimport mode
  -dynout string
        write -dynimport output to this file
  -dynpackage string
        set Go package for -dynimport output (default "main")
  -exportheader string
        where to write export header if any exported functions
  -gccgo
        generate files for use with gccgo
  -gccgopkgpath string
        -fgo-pkgpath option used with gccgo
  -gccgoprefix string
        -fgo-prefix option used with gccgo
  -godefs
        for bootstrap: write Go definitions for C file to standard output
  -import_runtime_cgo
        import runtime/cgo in generated code (default true)
  -import_syscall
        import syscall in generated code (default true)
  -importpath string
        import path of package being built (for comments in generated files)
  -objdir string
        object directory
  -srcdir string
        source directory
Use --strategy=CGoCodeGen=standalone to disable sandboxing for the failing actions.
INFO: Elapsed time: 1.771s, Critical Path: 1.47s
```

**Comment by jayconrod** [comment]
From the log, it looks like this was happening in a vendored `golang/x/sys/unix` package. Is that right?

**Comment by mintzhao** [comment]
yes, but I'm using command ```gazelle -external vendored -mode fix``` as BUILD file generator.

**Comment by jayconrod** [comment]
`golang.org/x/sys/unix` and `github.com/mattn/go-sqlite3` are packages with known issues that we haven't been able to build in the past. Hopefully #514 should fix this when it lands today.

Once that happens, please regenerate your BUILD files with gazelle, and see if the build works. If it's still broken, we can debug further.

If you have a repo where this is reproducible, please point to it. I tried `github.com/mintzhao/yoyo`, but it had other issues with missing dependencies, so maybe there are some commits missing.

**Comment by mintzhao** [comment]
ok, thanks. ```github.com/mintzhao/yoyo``` is fine, but you must *govendor sync* first.

**Comment by mintzhao** [comment]
well, new problem:
```
➜  yoyo git:(master) ✗ gazelle                                                           
2017/06/15 09:57:49 in dir "vendor/github.com/spf13/viper", could not resolve import path "gopkg.in/yaml.v2": unrecognized import path "gopkg.in/yaml.v2"
```

just used rule_go 0.5.0 and reinstall gazelle

**Comment by ianthehat** [comment]
This was fixed in #950
